xline/tests/watch_test.rs

```diff
@@ -0,0 +1,61 @@
+mod common;
+
+use std::error::Error;
+
+use etcd_client::EventType;
+use tracing::info;
+use xline::client::kv_types::{DeleteRangeRequest, PutRequest};
+
+use crate::common::Cluster;
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+async fn test_watch() -> Result<(), Box<dyn Error>> {
+    let mut cluster = Cluster::new(3).await;
+    cluster.start().await;
+    let client = cluster.client().await;
+    let mut watch_client = client.watch_client();
+
+    let (_watcher, mut stream) = watch_client.watch("foo", None).await?;
+    let handle = tokio::spawn(async move {
+        if let Ok(Some(res)) = stream.message().await {
+            let event = res.events().get(0).unwrap();
+            let kv = event.kv().unwrap();
+            assert_eq!(event.event_type(), EventType::Put);
+            assert_eq!(kv.key(), b"foo");
+            assert_eq!(kv.value(), b"bar");
+        }
+        if let Ok(Some(res)) = stream.message().await {
+            let event = res.events().get(0).unwrap();
+            let kv = event.kv().unwrap();
+            assert_eq!(event.event_type(), EventType::Delete);
+            assert_eq!(kv.key(), b"foo");
+            assert_eq!(kv.value(), b"");
+        }
+    });
+
+    client.put(PutRequest::new("foo", "bar")).await?;
+    client.delete(DeleteRangeRequest::new("foo")).await?;
+
+    handle.await?;
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+async fn test_watch_client_closes_connection() -> Result<(), Box<dyn Error>> {
+    let mut cluster = Cluster::new(3).await;
+    cluster.start().await;
+    let client = cluster.client().await;
+    let mut watch_client = client.watch_client();
+
+    let (watcher, mut stream) = watch_client.watch("foo", None).await?;
+    let handle = tokio::spawn(async move {
```

Comment:
How to tell the correctness of tests? How do we know the server handles the client close operation correctly?

Reply:
If we call this test before fixing it, the watch server will panic